sift/__main__.py

```diff
@@ -653,33 +707,15 @@ def __init__(self, config_dir=None, workspace_dir=None, cache_size=None, glob_pa
                                                parent=self)
         self.export_image = ExportImageHelper(self, self.document, self.scene_manager)
 
-        # connect canvas and projection pieces
-        self.ui.mainMapWidget.layout().addWidget(self.scene_manager.main_canvas.native)
-        self.ui.projectionComboBox.addItems(tuple(self.document.available_projections.keys()))
-        self.ui.projectionComboBox.currentIndexChanged.connect(self.document.change_projection_index)
-
-        self.scene_manager.didChangeFrame.connect(self.update_frame_slider)
-        self.ui.animPlayPause.clicked.connect(self.toggle_animation)
-        self.ui.animPlayPause.setContextMenuPolicy(QtCore.Qt.CustomContextMenu)
-        self.ui.animPlayPause.customContextMenuRequested.connect(self.show_animation_speed_slider)
-
-        def next_frame(*args, **kwargs):
-            self.scene_manager.layer_set.animating = False
-            self.scene_manager.layer_set.next_frame()
-
-        self.ui.animForward.clicked.connect(next_frame)
-
-        def prev_frame(*args, **kwargs):
-            self.scene_manager.layer_set.animating = False
-            self.scene_manager.layer_set.next_frame(frame_number=-1)
-
-        self.ui.animBack.clicked.connect(prev_frame)
-
-        # allow animation slider to set animation frame being displayed:
-        self.ui.animationSlider.valueChanged.connect(self.animation_slider_jump_frame)
+        self._init_layer_panes()
+        self._init_rgb_pane()
+        self._init_map_widget()
 
-        # allow animation, once stopped, to propagate visibility to the document and layerlist:
-        self.scene_manager.didChangeLayerVisibility.connect(self.document.animation_changed_visibility)
+        self.animation = UserControlsAnimation(self.ui,
```

Comment:
Is `animation` too ambiguous?

Reply:
I think it's okay, though animation_behavior might be appropriate - but would that turn into more questions?

Reply:
Possibly. Only other thought is if we want to have a naming convention for behaviors. I'm ok leaving it.